.github/workflows/typespec-validation-all.yaml

```diff
@@ -51,7 +55,7 @@ jobs:
 
       - uses: actions/checkout@v4
         with:
-          fetch-depth: 2
+          ref: ${{ matrix.ref != 'default' && matrix.ref || '' }}
```

Comment:
Any reason `${{ matrix.ref }}` doesn't work by itself? Does it not understand default and if not could we just say main?

Reply:
I added a comment explaining this.  Setting `ref` to an empty string (or not setting it at all), checks out the ref from the default context, which is usually what you want.  We don't want to ever hardcode `main`.

I **could** have put an empty string in the matrix, but the GitHub Jobs UI silently ignores empty strings, so using a placeholder like `"default"` made things clearer IMO.

Reply:
I added code comments explaining this.